R/getGrouping.R

```diff
@@ -0,0 +1,24 @@
+#' Get Data Grouping
+#'
+#' @param data input data.frame with variable to group
+#' @param grouping vector of variable names to group by
+#' @description 
+#' Function to group data and create groups according to variable
+#' So far, the grouping is performed only along one variable
+#' @return data with groups as factor
+#' @export
+#'
+#' @examples
+getGrouping <- function(data, grouping) {
+  # Check that grouping variable are in the data
+  stopifnot(grouping %in% names(data))
+  if (length(grouping)>1){stop("Grouping with more than one variable is not currently handled")}
+  
+  if (is.null(grouping)){

```

Comment:
Code can be simplified a bit. Good to try to always exit early if you can.
```
if(is.null(grouping)){
 return(factor(1))
}

return factor(data...))
```

Limiting the amount of else is crucial to simplify code

Reply:
Indeed, I set that at the very beginning, and was waiting for Abdullah's code to update.